source/extensions/http/cache/file_system_http_cache/cache_file_fixed_block.h

```diff
@@ -0,0 +1,121 @@
+
+  // cacheVersionId is a value that should be consistent between versions of the file
+  // cache implementation. Changing version in code will invalidate all cache
+  // entries where the version ID does not match.
+  uint32_t cacheVersionId() const {
+    return fromEndianness<ByteOrder::BigEndian>(contents_.cache_version_id_);
+  }
+
+  // Size of the serialized proto message capturing headers and metadata.
+  size_t headerSize() const { return fromEndianness<ByteOrder::BigEndian>(contents_.header_size_); }
+
+  // Size of the serialized proto message capturing trailers.
+  size_t trailerSize() const {
+    return fromEndianness<ByteOrder::BigEndian>(contents_.trailer_size_);
+  }
+
+  // Size of the http body of the cache entry.
+  size_t bodySize() const { return fromEndianness<ByteOrder::BigEndian>(contents_.body_size_); }
+
+  // Sets the size of the serialized headers, key and metadata.
+  void setHeadersSize(size_t sz) {
+    contents_.header_size_ = toEndianness<ByteOrder::BigEndian>(static_cast<uint32_t>(sz));
+  }
+
+  // Sets the size of the serialized trailers.
+  void setTrailersSize(size_t sz) {
+    contents_.trailer_size_ = toEndianness<ByteOrder::BigEndian>(static_cast<uint32_t>(sz));
+  }
+
+  // Sets the size of the serialized body.
+  void setBodySize(size_t sz) {
+    contents_.body_size_ = toEndianness<ByteOrder::BigEndian>(static_cast<uint64_t>(sz));
+  }
+
+  // Returns the offset from the start of the file to the start of the
+  // serialized headers proto.
+  static size_t offsetToHeaders() { return size(); }
+
+  // Returns the offset from the start of the file to the start of the body data.
+  size_t offsetToBody() const { return offsetToHeaders() + headerSize(); }
+
+  // Returns the offset from the start of the file to the start of the
+  // serialized trailers proto.
+  size_t offsetToTrailers() const { return offsetToBody() + bodySize(); }
+
+  // Returns a string_view of the serialized fixed header chunk.
+  absl::string_view stringView() const { return {contents_.as_str_, size()}; }
+
+  // Returns true if the fileId and cacheVersionId match the compile-time constants from
+  // cache_file_fixed_block.cc
+  bool isValid() const;
+
+private:
+  // sets the fileId, a fixed value used to identify that this is a cache file.
+  // This is private so only the unit test can call it, as the file ID should be
+  // a compile-time constant in ordinary use.
+  void setFileId(uint32_t id) { contents_.file_id_ = toEndianness<ByteOrder::BigEndian>(id); }
+
+  // sets the cacheVersionId, a value that should be consistent between versions
+  // of the file cache implementation.
+  // This is private so only the unit test can call it, as the version ID should be
+  // a compile-time constant in ordinary use.
+  void setCacheVersionId(uint32_t id) {
+    contents_.cache_version_id_ = toEndianness<ByteOrder::BigEndian>(id);
+  }
+
+  union {
```

Comment:
You probably need to defend the position that you need to manually serialize/de-serialize all these fields including network byte-ordering hacks, rather than simply using protobuf serialization. Please put the rationale in a comment.

Actually I'm not clear myself on why you went this way. I was caught up mainly in the details of *how* you did this but I'm finally catching up on *why* :) 

To be clear: I'm not necessarily objecting to the design decision but it does need to be documented.

Reply:
It was documented in the proto file where the whole structure of the file is explained rather than the header alone, but okay.